web/src/features/time/TimeController.tsx

```diff
@@ -16,89 +16,94 @@ import { useIsBiggerThanMobile } from 'utils/styling';
 
-  const onToggleGroupClick = useCallback(
-    (timeAverage: TimeAverages) => {
-      // Set time slider to latest value before switching aggregate to avoid flickering
-      setSelectedDatetime({
-        datetime: selectedDatetime.datetime,
-        index: numberOfEntries,
-      });
-      setTimeAverage(timeAverage);
-      trackEvent('Time Aggregate Button Clicked', { timeAverage });
-    },
-    [selectedDatetime.datetime, numberOfEntries, setSelectedDatetime, setTimeAverage]
-  );
+    const onTimeSliderChange = useCallback(
+      (index: number) => {
+        if (!datetimes) {
+          return;
+        }
+        setSelectedDatetime({
+          datetime: datetimes[index],
+          index,
+        });
+      },
+      [datetimes, setSelectedDatetime]
+    );
 
-  return (
-    <div className={twMerge(className, 'flex flex-col gap-3')}>
-      {isBiggerThanMobile && <TimeHeader />}
-      <TimeAverageToggle
-        timeAverage={timeAverage}
-        onToggleGroupClick={onToggleGroupClick}
-      />
-      <div>
-        {/* The above div is needed to treat the TimeSlider and TimeAxis as one DOM element */}
-        <TimeSlider
-          onChange={onTimeSliderChange}
-          numberOfEntries={numberOfEntries}
-          selectedIndex={selectedDatetime.index}
-        />
-        <TimeAxis
-          datetimes={datetimes}
-          selectedTimeAggregate={timeAverage}
-          isLoading={isLoading}
-          className="h-[22px] w-full overflow-visible"
-          transform={`translate(12, 0)`}
-          isLiveDisplay={isHourly}
+    const onToggleGroupClick = useCallback(
+      (timeAverage: TimeAverages) => {
+        setSelectedDatetime({
+          datetime: selectedDatetime.datetime,
+          index: numberOfEntries,
+        });
+        setTimeAverage(timeAverage);
+        trackEvent('Time Aggregate Button Clicked', { timeAverage });
+      },
+      [selectedDatetime.datetime, numberOfEntries, setSelectedDatetime, setTimeAverage]
+    );
+
+    return (
+      <div ref={reference} className={twMerge(className, 'flex flex-col gap-3')}>
+        {isBiggerThanMobile && <TimeHeader />}
+        <TimeAverageToggle
+          timeAverage={timeAverage}
+          onToggleGroupClick={onToggleGroupClick}
         />
+        <div>
+          {/* The above div is needed to treat the TimeSlider and TimeAxis as one DOM element */}
+          <TimeSlider
+            onChange={onTimeSliderChange}
+            numberOfEntries={numberOfEntries}
+            selectedIndex={selectedDatetime.index}
+          />
+          <TimeAxis
+            datetimes={datetimes}
+            selectedTimeAggregate={timeAverage}
+            isLoading={isLoading}
+            className="h-[22px] w-full overflow-visible"
+            transform={`translate(12, 0)`}
+            isLiveDisplay={isHourly}
+          />
+        </div>
       </div>
-    </div>
-  );
-}
+    );
+  }
+);
+
+TimeController.displayName = 'TimeController';
```

Comment:
If you use a function declaration inside the forward ref this is not needed.

With the changes in react 19 with ref as a prop we won't need forward ref anymore so it will be a cleaner change then if we used the fucntion declaration syntax instead.